src/components/pages/Profile/History/ComputeJobs/Results.tsx

```diff
@@ -40,74 +41,80 @@ export default function Results({
   const [hasFetched, setHasFetched] = useState(false)
   const isFinished = job.dateFinished !== null
 
-  async function getResults() {
-    if (!account || !ocean || !job) return
+  useEffect(() => {
+    async function getResults() {
+      if (!account || !ocean || !job) return
+
+      try {
+        setIsLoading(true)
+        const computeService = ddo.findServiceByType('compute')
+        const jobStatus = await ocean.compute.status(
+          account,
+          job.did,
+          ddo,
+          computeService,
+          job.jobId
+        )
+        if (jobStatus) {
+          job.results = jobStatus[0].results
+        }
+      } catch (error) {
+        Logger.error(error.message)
+      } finally {
+        setIsLoading(false)
+        setHasFetched(true)
+      }
+    }
 
+    getResults()
+  }, [ocean, account, job, ddo])
+
+  async function accessResult(result: ComputeResult, index: number) {
+    if (!account || !ocean || !job) return
     try {
-      setIsLoading(true)
       const computeService = ddo.findServiceByType('compute')
-      const jobStatus = await ocean.compute.status(
+      const jobResult = await ocean.compute.getResult(
```

Comment:
thought(non-blocking): I somewhat dislike functions that take 3+ arguments because you have to provide them one by one. We could use an array here and then pass the arguments using `Function.prototype.apply()`. I am not sure if we might lose any IntelliSense advantages here, though (need to check), but the code readability might improve. If there is no IntelliSense support, we might have issues updating the code and overloading the function without any feedback. 

Alternative: Create a helper function for `ocean.compute()` that expects an object as an argument and passes the single values inside the helper function. This would avoid difficult-to-read (jumping eye) code blocks, improve readability, not break any existing implementation, and provide IntelliSense support.

Food for thought, not blocking.

Reply:
I agree, we can have a look at this at some point. In this case we are reusing existing functions from Ocean Market, we would need to reimplement in such cases. Maybe it is also enough to introduce these rules for ower own codelines, but I do agree that it could imrpove readability.